src/_static/js/workflow.js

```diff
@@ -0,0 +1,72 @@
+class WorkflowContainer {
+    static cellCounter = 0;
+
+    static escapeHtml(unsafe)
+    {
+        return unsafe
+             .replace(/&/g, "&amp;")
+             .replace(/</g, "&lt;")
+             .replace(/>/g, "&gt;")
+             .replace(/"/g, "&quot;")
+             .replace(/'/g, "&#039;");
+     }
+
+    static createWorkflowCell(img, path) {
+        const wrap = document.createElement("div");
+        wrap.classList.add("highlight");
+
+        // workflow SVG renders in a visible pre element
+        const workflowCell = document.createElement("pre");
+        wrap.appendChild(workflowCell);
+        const imgParent = img.parentElement;
+        workflowCell.appendChild(img);
+        imgParent.remove();
```

Comment:
> > The rst equivalent for the current custom container syntax would be:
> 
> I actually have a question about the rst syntax. Have you tried whether the following would work?
> 
> ```
> .. container:: workflow
>
>     .. image:: ../workflows/CentroidModel.bonsai
>         :alt: Alternative text
> ```

Actually, the above would work if we change this line as follows:

```suggestion
        // remove the parent p element, if it exists
        if (imgParent.nodeName.toLowerCase() === "p") {
            imgParent.remove();
        }
```
Because in .md the custom container places the `<img>` within `<p>`, i.e. `<div><p><img>...</img></p></div>`, whereas in .rst we'd get `<div><img>...</img></div>`, and removing the parent element of the img means we remove the entire `<div>`.

Reply:
Happy to merge this PR once the above is resolved, and perhaps revisit the syntax discussion later. Nonetheless, it's nice to know the reasoning behind the chosen syntax, as I thought of it more as "code that one can copy", than a special kind of image that needs to be converted.

Reply:
@glopesdev in case you missed this 😄